special-pages/pages/new-tab/app/activity/mocks/activity.mock-transport.js

```diff
@@ -36,6 +36,17 @@ export function activityMockTransport() {
             /** @type {import('../../../types/new-tab.ts').NewTabMessages['notifications']} */
             const msg = /** @type {any} */ (_msg);
             switch (msg.method) {
+                case 'activity_removeItem': {
+                    const oldCount = dataset.activity.reduce((acc, item) => acc + item.trackingStatus.totalCount, 0);
+                    dataset.activity = dataset.activity.filter((x) => x.url !== msg.params.url);
+                    const cb = subs.get('activity_onDataPatch');
+                    const p = toPatch(dataset.activity);
+                    p.totalTrackersBlocked = oldCount;
+                    setTimeout(() => {
+                        cb(p);
+                    }, 0);
+                    break;
+                }
```

Comment:
This just allows the in-browser testing scenario. It models perfectly how the events occur on windows.

`activity_removeItem` notification is immediately followed by the browser pushing `activity_onDataPatch` into the page. This was a missing scenario in the mocks - not surprising then that a bug crept in!